### ISSUE TITLE
Magento 2 EE Catalog Rule Product / Catalog Product Rule indexer stuck processing

### ISSUE THREAD

**Issue by logan-robbins** [magento/magento2#2596]
Magento 22 EE with "large" profile for test data. After reindex, the Catalog Rule Product and Catalog Product Rule indexers are stuck in "Processing" in the admin panel. I reindexed just those 2 and cleared cache, but still says processing in admin panel.

**Comment by mazhalai** [comment]
@ljrweb please post EE related issues via the appropriate portal. This is only for CE related issues.

**Comment by iMilnb** [comment]
Sorry to re-open this subject but I also am witnessing this behavior, with CE version 2.0.5:

```
$ sudo -u ${wwwuser} magento indexer:status
Customer Grid:                                     Processing
Category Products:                                 Ready
Product Categories:                                Ready
Product Price:                                     Ready
Product EAV:                                       Ready
Stock:                                             Ready
Catalog Search:                                    Processing
Catalog Rule Product:                              Processing
Catalog Product Rule:                              Ready
```

The status is frozen in "Processing" for about 5 hours now. There's no MySQL queries running:

```
mysql> show full processlist;
+----+----------+---------------------+------+---------+------+----------+-----------------------+
| Id | User     | Host                | db   | Command | Time | State    | Info                  |
+----+----------+---------------------+------+---------+------+----------+-----------------------+
|  2 | rdsadmin | localhost:47537     | NULL | Sleep   |    9 |          | NULL                  |
| 51 | magento  | 1.1.1.1:49044 | NULL | Query   |    0 | starting | show full processlist |
+----+----------+---------------------+------+---------+------+----------+-----------------------+
2 rows in set (0.00 sec)
```

Any hint where I should start looking?

**Update**

I found a workaround here http://magento.stackexchange.com/questions/96590/magento-2-0-with-upgraded-to-php-7-0-indexer-stuck-at-processing

```
mysql> update magento.indexer_state set status='invalid' where status ='working';
Query OK, 3 rows affected (0.00 sec)
Rows matched: 3  Changed: 3  Warnings: 0
mysql> Bye

$ sudo -u ${wwwuser} magento indexer:status
Customer Grid:                                     Reindex required
Category Products:                                 Ready
Product Categories:                                Ready
Product Price:                                     Ready
Product EAV:                                       Ready
Stock:                                             Ready
Catalog Search:                                    Reindex required
Catalog Rule Product:                              Reindex required
Catalog Product Rule:                              Ready

$ sudo -u ${wwwuser} magento indexer:reindex
Customer Grid index has been rebuilt successfully in 00:00:00
Category Products index has been rebuilt successfully in 00:00:00
Product Categories index has been rebuilt successfully in 00:00:00
Product Price index has been rebuilt successfully in 00:00:00
Product EAV index has been rebuilt successfully in 00:00:00
Stock index has been rebuilt successfully in 00:00:00
Catalog Search index has been rebuilt successfully in 00:00:01
Catalog Rule Product index has been rebuilt successfully in 00:00:02
Catalog Product Rule index has been rebuilt successfully in 00:00:02

$ sudo -u ${wwwuser} magento indexer:status
Customer Grid:                                     Ready
Category Products:                                 Ready
Product Categories:                                Ready
Product Price:                                     Ready
Product EAV:                                       Ready
Stock:                                             Ready
Catalog Search:                                    Ready
Catalog Rule Product:                              Ready
Catalog Product Rule:                              Ready
```

Nevertheless, how can this happen with a stock installation?

**Comment by maksek** [comment]
@iMilnb, here http://magento.stackexchange.com/a/96607/35714 possible explanation. Which PHP version you are using?

**Comment by iMilnb** [comment]
@maksek PHP 7.0.5 in FPM mode with nginx, but note that I did not witness segfaults, only frozen "Processing" status.

**Comment by maksek** [comment]
@iMilnb It may be related to the issue - https://bugs.php.net/bug.php?id=71914, also the issue with PHP7.0.5 mentioned here - https://github.com/magento/magento2/issues/4070
And i docs you can find - we are not recommending using 7.0.5 due the bugs, check PHP version section - http://devdocs.magento.com/guides/v2.0/install-gde/system-requirements.html.
Can you try to downgrade or upgrade PHP from 7.0.5 and check again?

**Comment by iMilnb** [comment]
@maksek I definitely missed the "do not use 7.0.5" part :( Unfortunately I can't downgrade without rebuilding my own packages as < 7.0.5 is not available in dotdeb, but 7.0.6 is now available so I've upgraded, we'll see how it goes. Thanks a lot for your feedback, this definitely seems to be the root cause, hope this thread will help others.
